ui/src/plugins/dev.perfetto.AndroidLog/logs_panel.ts

```diff
@@ -325,8 +361,45 @@ export class LogsFilters implements m.ClassComponent<LogsFiltersAttrs> {
         },
         disabled: attrs.store.state.textEntry === '',
       }),
+      ...(hasMachineIds ? [this.renderFilterPanel(attrs)] : []),
```

Comment:
`hasMachineIds && this.renderFilterPanel(attrs)` works just fine here - mithril omits falsy children for this exact reason.